### DISCUSSION TITLE
Data missing from old projects

### DISCUSSION THREAD

**Discussion by tomhajjar** [ra3xdh/qucs_s#772] [Q&A] [answered]
I was revisiting and old project at the link below. The author updated the files to be ngspice compatible 2 years ago. 

I cannot reproduce the plotted data because some of the variables derived from Nutmeg equations are missing from the data file. It's not consistent what variables are missing, The screen shots show what the plots looked like prior to clicking "Simulate".

"gain" is always missing. vout_db is missing from some 

https://github.com/saul-rodriguez/QUCS_ANALOG_ELEKTRONIK
![2024-06-18_233911](https://github.com/ra3xdh/qucs_s/assets/102838604/167e6747-63b8-4940-8cbe-b6d69ac485e4)
![2024-06-18_233543](https://github.com/ra3xdh/qucs_s/assets/102838604/680f9aca-0d19-456a-946d-08279dedaa29)
![2024-06-18_231128](https://github.com/ra3xdh/qucs_s/assets/102838604/0be6b2b1-26ad-47bf-8d5e-e11fc7cc164e)
![2024-06-18_234207](https://github.com/ra3xdh/qucs_s/assets/102838604/2ded66e1-2b28-4295-afbf-98ad00ea33c0)

[IE1202_F13_prj.zip](https://github.com/user-attachments/files/15895215/IE1202_F13_prj.zip)

**Comment by ra3xdh** [comment, marked as answer]
It is not a bug. We have recently redesigned the nutmeg equations. See #463 and #519. You have to select a correct simulation from the drop-down list in the simulation properties. Also keep in mind that DC became a valid simulation for Ngspice. Please remove the `ac.` prefixes in variables. Ngspice will not accept it. After applying these corrections the schematic works as expected. 

![image](https://github.com/ra3xdh/qucs_s/assets/4920080/36bef52e-73c8-4843-b0cc-56e476ad864e)

**Reply by ra3xdh** [reply]
Also you should recreate the plots and select a correct variable name.

**Comment by tomhajjar** [comment]
I redid my files and got them to work except Transient wouldn't work. I then made some minor mods and all the plots but Vout disappeared. This has happened many times during editing but I can usually get the plots back. 

Netlist shows that Vout_dB and Vout_phase are missing from the "ac" write statement.

Netlist shows that Gain is missing from the "dc" write statement.

 Netlist shows Vin and Vout are in the "tran" write statement but no data is plotted.

Oddly plots have worked even when my Nutmeg syntax is wrong. I make a minor change in the schematic and plots disappear. I undo the change and the plots never come back. I think there is a bug causing my plots to appear and disappear. 

![2024-06-19_170910](https://github.com/ra3xdh/qucs_s/assets/102838604/206c939c-80ac-4094-9ea3-b1421b9ca6a2)
![2024-06-19_170954](https://github.com/ra3xdh/qucs_s/assets/102838604/a6ebba5a-3015-4128-b64b-8b700410a91a)

**Comment by ra3xdh** [comment]
> I think there is a bug causing my plots to appear and disappear.

There is no bug. Everything works as expected on my machine. You have assigned a wrong simulation type for gain. It should be `SW1` but not `DC`. The `DC` is operation point analysis. It is 1-point simulation and `deriv` is not valid for it. 

![image](https://github.com/ra3xdh/qucs_s/assets/4920080/6a6d8a95-3b36-49f9-8266-60ac609e4ad3)

**Comment by tomhajjar** [comment]
I accidently sent the wrong file above. Your answer doesn't explain why Nutmeg variables keep disappearing from the netlist after minor schematic changes that shouldn't effect anything.

I am still having issues with Qucs-S not saving data and plotting Nutmeg variables after making minor changes to the schematic. When I hand edit the netlists to add plot commands with the missing variables they work fine under DuSpice. 

Attached are three scenarios. Schematic unaltered. Cin deleted (Cin not used). UIN_DC renamed to VIN_DC. Attachment includes unaltered schematic and netlists edited for DuSpice. Cin and UIN_DC cases work in DuSpice but not Qucs-S.

Attached is a screen shot comparing unedited netlist to netlist with UIN_DC renamed to VIN_DC. Clearly variables AB_dB, AB_phase and Vout_dB are missing.

![2024-06-20_110207](https://github.com/ra3xdh/qucs_s/assets/102838604/09da8734-8efe-4ee8-9b70-4b80678e516d)

![2024-06-20_110321](https://github.com/ra3xdh/qucs_s/assets/102838604/16d7d4d8-7d56-4264-8297-ca787422a4fe)

![2024-06-20_110444](https://github.com/ra3xdh/qucs_s/assets/102838604/342e3209-be53-41ef-8734-975821ffb5cd)

![2024-06-20_111806](https://github.com/ra3xdh/qucs_s/assets/102838604/607aae8b-e33c-40f0-a70d-326e83d21b4e)

![2024-06-20_112218](https://github.com/ra3xdh/qucs_s/assets/102838604/471c0169-94c6-4ec6-b8c8-7157b228bfcd)

![2024-06-20_114005](https://github.com/ra3xdh/qucs_s/assets/102838604/064f4638-a9c6-4bb3-9f38-c9bc7e81f439)

[TWO_STAGE_AMP_SPICE.zip](https://github.com/user-attachments/files/15916481/TWO_STAGE_AMP_SPICE.zip)

**Reply by ra3xdh** [reply]
Don't mix Qucs legacy and Nutmeg equations in one ciruit. Qucs legacy equations has only limited support in SPICE mode and may cause issues. Something may have changed after #463. Its operation in SPICE mode was never warrantied. Everything works as expected after replacement of the Qucs equation by .PARAM.

![image](https://github.com/ra3xdh/qucs_s/assets/4920080/f18dcad9-37b9-4fbf-9666-09c37c8293f4)

**Comment by tomhajjar** [comment]
It's good to know I'm not going crazy. It was like chasing ghosts. A lot of schematics like this that were tested under previous versions of Qucs-S will need to be updated. 

It would be useful to figure out and fix why such minor changes to the schematic would cause these problems. Seems to me it has or will cause other issues. I've been seeing other issues occur that are hard to identify and document.
